File: sdk/ts/src/capabilities.ts
import { ClientCore } from "./core.js";
import type { CapabilitiesResponse } from "./types.js";

export class CapabilitiesClient {
  constructor(private readonly core: ClientCore) {}

  /** Get server capability limits for adaptive clients. */
  async get(): Promise<CapabilitiesResponse> {
    return this.core.doJSON<CapabilitiesResponse>("GET", "/v1/capabilities");
  }
}

File: sdk/filegate/capabilities.go
package filegate

import (
	"context"
	"net/http"
)

// CapabilitiesClient contains server capability endpoints.
type CapabilitiesClient struct {
	core *clientCore
}

func (c CapabilitiesClient) Get(ctx context.Context) (*CapabilitiesResponse, error) {
	var out CapabilitiesResponse
	if err := c.core.doJSON(ctx, http.MethodGet, "/v1/capabilities", nil, nil, "", &out); err != nil {
		return nil, err
	}
	return &out, nil
}
